Makefile

```diff
@@ -1,92 +1,60 @@
-# Copyright 2018 The Kubernetes Authors.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
 
-.PHONY: genapi genconversion genclientset gendeepcopy
+# Image URL to use all building/pushing image targets
+IMG ?= gcr.io/k8s-cluster-api/cluster-api-controller:latest
 
-all: generate build images
+all: test manager clusterctl
 
-depend:
-	dep version || go get -u github.com/golang/dep/cmd/dep
-	dep ensure -v
+# Run tests
+test: generate fmt vet manifests
+	go test -v -tags=integration ./pkg/... ./cmd/...
+#	go test -v -tags=integration ./pkg/... ./cmd/... -coverprofile cover.out
 
-	# go libraries often ship BUILD and BUILD.bazel files, but they often don't work.
-	# We delete them and regenerate them
-	find vendor -name "BUILD" -delete
-	find vendor -name "BUILD.bazel" -delete
+# Build manager binary
+manager: generate fmt vet
+	go build -o bin/manager sigs.k8s.io/cluster-api/cmd/manager
 
-	bazel run //:gazelle
+# Build manager binary
+clusterctl: generate fmt vet
+	go build -o bin/clusterctl sigs.k8s.io/cluster-api/cmd/clusterctl
 
-generate: genapi genconversion genclientset gendeepcopy genopenapi
+# Run against the configured Kubernetes cluster in ~/.kube/config
+run: generate fmt vet
+	go run ./cmd/manager/main.go
 
-genapi: depend
-	go build -o $$GOPATH/bin/apiregister-gen sigs.k8s.io/cluster-api/vendor/github.com/kubernetes-incubator/apiserver-builder/cmd/apiregister-gen
-	$$GOPATH/bin/apiregister-gen -i ./pkg/apis,./pkg/apis/cluster,./pkg/apis/cluster/v1alpha1
+# Install CRDs into a cluster
+install: manifests
+	kubectl apply -f config/crds
 
-genconversion: depend
-	go build -o $$GOPATH/bin/conversion-gen sigs.k8s.io/cluster-api/vendor/k8s.io/code-generator/cmd/conversion-gen
-	$$GOPATH/bin/conversion-gen -i ./pkg/apis/cluster/v1alpha1/ -O zz_generated.conversion --go-header-file boilerplate.go.txt
+# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
+deploy: manifests
+	kubectl apply -f config/crds
```

Comment:
Nit: this line could be replaced by `deploy: manifests install`.

Reply:
I am not sure the CRDs need to be installed separately since they are referenced by `config/default/kustomization.yaml`. Am I missing something?

Reply:
No.  That can probably just be deleted.